How was empython built

Hi @aidanhs 
I am curious on how empython was built. I also noticed I could not run the code on mac OS X.

Here I got:

```
$ make -f ../Makefile prep
(...)
gcc -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes  Parser/acceler.o Parser/grammar1.o Parser/listnode.o Parser/node.o Parser/parser.o Parser/parsetok.o Parser/bitset.o Parser/metagrammar.o Parser/firstsets.o Parser/grammar.o Parser/pgen.o Objects/obmalloc.o Python/mysnprintf.o Python/pyctype.o Parser/tokenizer_pgen.o Parser/printgrammar.o Parser/pgenmain.o -ldl  -framework CoreFoundation -o Parser/pgen
make[1]: Nothing to be done for `python'.
#cp Makefile ../Makefile.native
#cp Parser/pgen ../pgen.native
cp python ../python.native
cp: python is a directory (not copied).
make: *** [prep] Error 1
```

On OSX Mavericks. Anyone with better luck on other systems?

--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ EMEXPORTS=\
 	-s EXPORTED_FUNCTIONS="['_Py_Initialize', '_PyRun_SimpleString']" -s "EXTRA_EXPORTED_RUNTIME_METHODS=['cwrap']"
 
 empython.js: python/libpython3.5.a
-	python2 mapfiles.py python/Lib datafilezip > js/postJs.js
+	./mapfiles.py python/Lib datafilezip > js/postJs.js
 	cat js/postJs.js.in >> js/postJs.js
 	emcc $(EMFLAGS) $(EMEXPORTS) -o $@ $< python/Modules/zlib/libz.a
 

--- a/README.md
+++ b/README.md
@@ -1,3 +1,18 @@
+= empython
+
+== Get it
+
+First, clone and initialise submodules:
+
+```
+$ git clone git@github.com:aidanhs/empython.git
+$ cd empython
+$ git submodule update --init
+```
+
+== Build it
+
+If you have emscripten 2.0.7 installed (other versions may or may not work):
 
 ```
 $ cd python
@@ -5,4 +20,14 @@ $ make -f ../Makefile prep
 $ make -f ../Makefile em
 $ cd ..
 $ make empython.js
+```
+
+If you have docker or podman (below I'll use podman, I think you'll need to add `-u $(id -u):$(id -g)` for Docker):
+
+```
+$ podman pull emscripten/emsdk:2.0.7
+$ podman run -it --rm -v $(pwd):/src emscripten/emsdk:2.0.7 bash -c "cd python && make -f ../Makefile prep && make -f ../Makefile em && cd .. && make empython.js"
+$ ls -l empython.{js,wasm}
+.rw-r--r-- 2.9M aidanhs 25 Oct 14:52 empython.js
+.rwxr-xr-x 1.9M aidanhs 25 Oct 14:52 empython.wasm
 ```

--- a/mapfiles.py
+++ b/mapfiles.py
@@ -1,11 +1,11 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 from __future__ import print_function
 
 import os
 import sys
 import py_compile
 from zipfile import ZipFile, ZIP_DEFLATED
-from StringIO import StringIO
+from io import BytesIO
 from subprocess import check_call
 import base64
 
@@ -17,7 +17,7 @@ def mk_contents(data):
         'for(i=0;i<%s;i++)a[i]=t.charCodeAt(i);'
         'return a'
         '})()'
-    ) % (base64.b64encode(data), len(data), len(data))
+    ) % (base64.b64encode(data).decode('ascii'), len(data), len(data))
 
 def files_to_datafilecalls(fpaths):
     basedir = '/usr/local/lib/python3.5'
@@ -44,7 +44,7 @@ def files_to_datafilecalls(fpaths):
     return commands
 
 def files_to_datafilezipcall(fpaths):
-    zf = StringIO()
+    zf = BytesIO()
     zipfile = ZipFile(zf, 'w', ZIP_DEFLATED)
     for fpath, targetdir in fpaths:
         zipfile.write(fpath, os.path.join(targetdir, os.path.basename(fpath)))

